app/src/main/java/com/nextcloud/talk/utils/PushUtils.java

```diff
@@ -128,14 +133,13 @@ public SignatureVerification verifySignature(byte[] signatureBytes, byte[] subje
 
     private int saveKeyToFile(Key key, String path) {
         byte[] encoded = key.getEncoded();
-        FileOutputStream keyFileOutputStream = null;
-        try {
+
+        try (FileOutputStream keyFileOutputStream = new FileOutputStream(path)) {
+            keyFileOutputStream.write(encoded);
```

Comment:
Hm, shouldn't this go AFTER creating a file?

Reply:
Agreed. Also, we should probably ensure we could actually create the file by checking the return value.

Reply:
Yes, in theory, but it should mostly work because it's internal storage? :) But feel free to add additional safeguards! :)

Reply:
done

Reply:
@ardevd you pushed a wrong commit? :)